### ISSUE TITLE
Tag format is not valid ios/beta/1.0.0/10

### ISSUE THREAD

**Issue by aminerol** [xotahal/fastlane-plugin-semantic_release#27]
This tag format pattern `[platform]/[flavor]/[semver]/[build_number] `is not valid anymore, when i `excute analyze_comits` action, it returns this error.

`Error while parsing version from tag  by using tag_version_match - \d+\.\d+\.\d+. Please check if the tag contains version as you expect and if you are using single brackets for tag_version_match parameter.`

**Comment by xotahal** [comment]
@aminerol What version do you have?

**Comment by paulineramos** [comment]
I was getting the same error, and had to downgrade to v1.9.1.

**Comment by hanno-jonlay** [comment]
I saw this when I ran the command while I had no commits since my last `ios/alpha/0.1.0/3` tag.

I added a new commit with `fix: some title here`, and was then able to successfully build.

So my suspicion is that this is perhaps a misleading error message.

(Sorry to not have the time right now to make a repro/debug fully)

**Comment by xotahal** [comment]
Hey @aminerol 👋  Why did you close this? Did you find a fix? If so, can you share it with us? Thanks 🙏

**Comment by aminerol** [comment]
Hey, sorry for the late reply, as for the version i was using the latest version.
after reporting the issue i noticed after adding other commits the issue disappear even without removing the tag.
i will try to reproduce this issue with new repo

**Comment by janziemba** [comment]
@xotahal It happened to me when I had brackets () in the last commit. New commit "fixed" it. 🙂
Version: 1.12.0

**Comment by Psique** [comment]
Hi there,

We have recently encountered the stated error as well. We figured out that the [following piece of code](https://github.com/xotahal/fastlane-plugin-semantic_release/blob/bc5ee6732251eb5c9866d34687f6a4b4c04c5b30/lib/fastlane/plugin/semantic_release/actions/analyze_commits.rb#L62) returns empty string when the tag version has the format `vX.Y.Z`.

We included to our tag one hyphen, this is `vX.Y.Z-0`, and still returns an empty string. It was only when following the `vX.Y.Z-0-1` format with two hyphens that we were able to avoid this error.

We encourage the developers to allow versioning of tags such as `vX.Y.Z` without any hyphens.

Hope this help. Cheers.
